src/systems/diff_drive/DiffDrive.cc

```diff
@@ -187,18 +187,14 @@ void DiffDrive::Configure(const Entity &_entity,
     return;
   }
 
-  // Ugly, but needed because the sdf::Element::GetElement is not a const
-  // function and _sdf is a const shared pointer to a const sdf::Element.
-  auto ptr = const_cast<sdf::Element *>(_sdf.get());
-
   // Get params from SDF
-  sdf::ElementPtr sdfElem = ptr->GetElement("left_joint");
+  sdf::ElementConstPtr sdfElem = _sdf->FindElement("left_joint");
```

Comment:
I think this is actually a good use case for `auto`.  We shouldn't really care what the type is and the compiler should enforce the const correctness.

Reply:
ok, I'll update this and similar types to `auto`

Reply:
I had a fever, and the prescription was more `auto`

https://github.com/gazebosim/gz-sim/pull/2231/commits/7b6e966246d3307326e7af367cc6e37073e2ec96